tests/TransformTest.cpp

```diff
@@ -0,0 +1,245 @@
+#include "app_helper.hpp"
+
+#include <unordered_map>
+
+TEST_F(TApp, SimpleTransform) {
+    int value;
+    auto opt = app.add_option("-s", value)->transform(CLI::Transformer({{"one", std::string("1")}}));
+    args = {"-s", "one"};
+    run();
+    EXPECT_EQ(1u, app.count("-s"));
+    EXPECT_EQ(1u, opt->count());
+    EXPECT_EQ(value, 1);
+}
+
+TEST_F(TApp, SimpleTransformInitList) {
+    int value;
+    auto opt = app.add_option("-s", value)->transform(CLI::Transformer({{"one", "1"}}));
+    args = {"-s", "one"};
+    run();
+    EXPECT_EQ(1u, app.count("-s"));
+    EXPECT_EQ(1u, opt->count());
+    EXPECT_EQ(value, 1);
+}
+
+TEST_F(TApp, SimpleNumericalTransform) {
+    int value;
+    auto opt = app.add_option("-s", value)->transform(CLI::Transformer(CLI::TransformPairs<int>{{"one", 1}}));
+    args = {"-s", "one"};
+    run();
+    EXPECT_EQ(1u, app.count("-s"));
+    EXPECT_EQ(1u, opt->count());
+    EXPECT_EQ(value, 1);
+}
+
+TEST_F(TApp, EnumTransform) {
+    enum class test : int16_t { val1 = 3, val2 = 4, val3 = 17 };
+    test value;
+    auto opt = app.add_option("-s", value)
+                   ->transform(CLI::Transformer(
+                       CLI::TransformPairs<test>{{"val1", test::val1}, {"val2", test::val2}, {"val3", test::val3}}));
+    args = {"-s", "val1"};
+    run();
+    EXPECT_EQ(1u, app.count("-s"));
+    EXPECT_EQ(1u, opt->count());
+    EXPECT_EQ(value, test::val1);
+
+    args = {"-s", "val2"};
+    run();
+    EXPECT_EQ(value, test::val2);
+
+    args = {"-s", "val3"};
+    run();
+    EXPECT_EQ(value, test::val3);
+
+    args = {"-s", "val4"};
+    EXPECT_THROW(run(), CLI::ConversionError);
```

Comment:
Add `args = {"-s", "5"};` here too

Reply:
That won't produce any errors, since Transformer explicitly doesn't do any checking, and 5 is still a valid member of the underlying type.  Is the proposed check just to verify that?

Reply:
Yes. Tests for things that pass are good too, plus it's an example of what's different.